segments-explainer/SEGMENTS-EXPLAINER.md

```diff
@@ -0,0 +1,100 @@
+
+```
+interface VisualViewport : EventTarget {
+	readonly attribute FrozenArray<DOMRect> segments;
+}
+```
+
+The value returned from querying `visualViewport.segments` will be an array of DOMRects and based on the data returned for each "viewport segment", developers will be able to infer the number of hinges available as well as the hinge orientation. When `visualViewport.segments` is queried on a desktop, this will return `null` since there is only one viewport segment.
+
+A user may at any point take the browser window out of spanning mode and place it on one of the screens or vice-versa, in those cases the window resize event will fire and authors can query and get the number of available viewport segments.
+
+This proposal doesn't aim to substitute existing APIs &mdash; the proposed development model can be summarized as requesting current viewport segments on interesting events and adjusting to the new presentation environment. There are no additional lifecycle proposals - the viewport segments are immutable and developers would request them upon common sense events (e.g. orientationchange, resize). It also  doesn't suggest how developers would use viewport segments to position, scale and orient content - in practical explorations developers used viewport segments to select the best declarative layout, not to modify layouts in script, but either would be possible.
+
+## Security and Privacy
+
+### APIs avalibility in iframe context
+
+`segments` will return null when called from within an `iframe` context.
+
+## Examples of user experiences and solution outlines that can leverage two screens:
+
+Let's take a look at a few practical examples of the scenarios above and how viewport segments would allow to resolve them for better user experience. In each case we'll start with some existing scenario and complicate it to provide opportunity to apply the proposal.
+
+### A map application that presents a map on one window segment and search results on another
+
+![Foldable with the left segment of the window containing a map and the right segment containing list of search results](/segments-explainer/map-app.svg)
+
+#### JavaScript solution outline:
+
+```js  
+const segments = window.visualViewport.segments;
+
+if( segments.length > 1 ) {
+	// now we know the device is a foldable
+	// it's recommended to test whether segments[0].width === segments[1].width
+	// and we can update CSS classes in our layout as appropriate 
```

Comment:
Can we link to the CSS specs (and/or give some examples) to tie these concepts together?